src/sequelize.js

```diff
@@ -333,6 +277,22 @@ export class Sequelize {
       dialectOptions: this.options.dialectOptions,
     };
 
+    // Convert replication connection strings to objects
+    if (this.options.replication) {
+      if (this.options.replication.write && typeof this.options.replication.write === 'string') {
+        this.options.replication.write = this.parseConnectionString(this.options.replication.write);
+      }
+
+      if (this.options.replication.read && this.options.replication.read.length) {
+        for (const i in this.options.replication.read) {
```

Comment:
This rule is not running on JS files yet but we're banning `for-i` due to it including non-own properties. Use this instead:

```suggestion
        for (let i = 0; i < this.options.replication.read.length; i++) {
```